jax/interpreters/xla.py

```diff
@@ -533,6 +602,52 @@ def _xla_callable(fun: lu.WrappedFun, device, backend, name, *arg_specs):
   else:
     return partial(_execute_replicated, compiled, result_handlers)
 
+def configure_aliasing(
+    c: xc.XlaBuilder,
+    xla_args: Sequence[xe.XlaOp],
+    out_tuple: xe.XlaOp,
+    arg_donate: Sequence[bool],
+    tuple_args: bool,
+):
+  """Configures input/output "must" aliasing based on `arg_donate`."""
+  assert len(xla_args) == len(arg_donate)
+
+  # First for every input array add it to `donations` iff it is a member of
+  # `arg_donate`.
+  donations = defaultdict(deque)
```

Comment:
Maybe a type hint or comment would help?

Reply:
I think this method is pretty clear now, adding types would probably complicate the code a fair bit and not add too much value.